0000-deb-repo-sync-gpg-check.md

```diff
@@ -0,0 +1,178 @@
+- Feature Name: deb_repo_sync_gpg_check
+- Start Date: 12-06-2019
+- RFC PR: (leave this empty)
+
+# Summary
+
+Verify GPG signature of Debian repositories when synchronizing.
+
+# Motivation
+
+- Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Uyuni can synchronize Debian repositories but currently it doesn't verify the GPG signature of the repo metadata.
+In order to import the metadata in a secure way Uyuni must be able to verify the GPG signature of that metadata.
+
+# Detailed design
+
+## Database changes
+
+A new table is needed to store the association of a GPG key to a repository:
```

Comment:
Not sure why we need this. We do already GPG metadata signature checking for SUSE repos and we do not need such an extra table. Why should we need it for debian?

Reply:
The idea would be to let user add GPG keys for third-party repos.

Under `Software -> Manage -> Repositories` UI there's the option to select SSL certificates for checking and/or authenticating the HTTPS URL or the repo. The SSL certs are defined under `Systems -> Autoinstallation -> GPG and SSL keys` (indeed not very intuitive). This allows uploading SSL certs or GPG keys but there's currently no way to link the GPG key to a repo (only SSL certs can be linked to a repo), hence this new table.

Reply:
Oh, these are used to access RedHat repos. They do client certificiate authentication.
repo-sync access these values and expect a X509 SSL certificate.

Reply:
There are definitely Debian repos that guard with X.509 certificates. Many of the ones I have to deal with internally are set up that way.

Reply:
Do you have an example repo which is not signed with GPG but with X509 ?

Reply:
Not one you can publicly access, unfortunately.